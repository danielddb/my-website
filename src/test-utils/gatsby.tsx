import { FixedObject } from 'gatsby-image';

export const createGatsbyFixedImageMock = (): FixedObject => ({
  base64:
    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z',
  width: 64,
  height: 64,
  src: '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg',
  srcSet:
    '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x'
});
