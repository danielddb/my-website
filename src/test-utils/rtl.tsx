import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import light from '../themes/light';

const Wrapper = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
