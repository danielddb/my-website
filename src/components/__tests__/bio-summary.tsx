import { useStaticQuery } from 'gatsby';
import { axe } from 'jest-axe';
import React from 'react';
import { createGatsbyFixedImageMock } from 'test-utils/gatsby';
import { render } from 'test-utils/rtl';
import Avatar from '../avatar';
import BioSummary from '../bio-summary';

const mockStaticQuery = useStaticQuery as jest.Mock<any>;

beforeEach(() => {
  mockStaticQuery.mockImplementation(() => ({
    placeholderImage: {
      childImageSharp: {
        fixed: createGatsbyFixedImageMock()
      }
    }
  }));
});

test('it should render correctly', () => {
  const { container } = render(<BioSummary />);

  expect(container).toMatchSnapshot();
});

test('it should render with default options', () => {
  const { getByAltText, getByText } = render(<BioSummary />);
  expect(
    getByText(/dan dawson-brown/i, { selector: 'h5' })
  ).toBeInTheDocument();
  expect(
    getByText(/ui developer from brum/i, { selector: 'h6' })
  ).toBeInTheDocument();
  expect(getByAltText(/dan dawson-brown/i)).toBeInTheDocument();
});

test('it should render with custom options', () => {
  const title = 'Custom title';
  const subtitle = 'Custom subtitle';
  const altText = 'Custom alt text';
  const { getByAltText, getByText } = render(
    <BioSummary
      title={title}
      subtitle={subtitle}
      avatar={<Avatar fixed={createGatsbyFixedImageMock()} alt={altText} />}
    />
  );
  expect(getByText(title, { selector: 'h5' })).toBeInTheDocument();
  expect(getByText(subtitle, { selector: 'h6' })).toBeInTheDocument();
  expect(getByAltText(altText)).toBeInTheDocument();
});

test('it should render with no accessibility issues', async () => {
  const { container } = render(<BioSummary />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
