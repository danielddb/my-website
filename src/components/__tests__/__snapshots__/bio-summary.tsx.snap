// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
.c2 {
  display: block;
  border-radius: 50%;
  overflow: hidden;
  border: 0.1875rem solid #e7e7e7;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  margin-right: 1rem;
}

.c3 {
  margin: 2rem 0 0;
  font-size: 1.40625rem;
  line-height: 2rem;
  margin-top: 0 !important;
}

.c4 {
  margin: 2rem 0 0;
  font-size: 1.125rem;
  line-height: 2rem;
  margin-top: 0 !important;
  color: #888888;
  font-weight: normal;
  margin-top: 0 !important;
}

@media (min-width:62em) {
  .c3 {
    margin: 3rem 0 0;
  }
}

@media (min-width:62em) {
  .c4 {
    margin: 3rem 0 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 gatsby-image-wrapper"
        style="position: relative; overflow: hidden; display: inline-block; width: 64px; height: 64px;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
        />
        <picture>
          <source
            srcset="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x"
          />
          <img
            alt="Author - Dan Dawson-Brown - UI Developer from Brum"
            height="64"
            loading="lazy"
            src="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg"
            srcset="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            width="64"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x" /&gt;&lt;img loading="lazy" width="64" height="64" srcset="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,
/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x" src="/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg" alt="Author - Dan Dawson-Brown - UI Developer from Brum" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
    <div>
      <h5
        class="sc-AxirZ c3"
      >
        Dan Dawson-Brown
      </h5>
      <h6
        class="sc-AxirZ c4"
      >
        UI Developer from Brum building tech for Vermeg.
      </h6>
    </div>
  </div>
</div>
`;
